<?xml version="1.0"?>
<launch>

    <arg name="save_posegraph" default="false" />
    
    <!-- For Velodyne VLP-16 -->
    <param name="scan_line" value="16" />
    

    <!-- For Velodyne HDL-32 
    <param name="scan_line" value="32" />
    -->

    <!-- For Velodyne HDL-64 
    <param name="scan_line" value="64" />
    -->
    <param name="directory_output" value="/home/$(env USER)/Downloads/export/" />
    <!--- Sim Time -->
    <param name="scan_period" value="0.1" />

    <param name="vertical_angle" type="double" value="2.0" />
    <param name="map_resolution" type="double" value="0.5" />
    <param name="output_downsample_size" type="double" value="0.3" />
    <param name="deskew" type="bool" value="true" />

    <param name="imu_topic" type="string" value="/nav/filtered_imu/data" /> <!-- Structor FLOAM -->
    <!--param name="imu_topic" type="string" value="/imu/data" /-->  <!-- Structor FLOAM -->

    <param name="loss_function" type="string" value="Cauchy" />
    <!--param name="loss_function" type="string" value="Cauchy" /-->  <!-- Structor FLOAM -->

    <param name="save_BALM" type="bool" value="true" />
    <param name="save_Posegraph" type="bool" value="$(arg save_posegraph)" />



    <param name="max_dis" type="double" value="90.0" />
    <param name="min_dis" type="double" value="0.2" />
    <!--- -->
    <node pkg="floam" type="floam_odom_estimation_node" name="floam_odom_estimation_node" output="screen"/>
    <node pkg="floam" type="floam_laser_processing_node" name="floam_laser_processing_node" output="screen"/>
    <node pkg="floam" type="floam_laser_mapping_node" name="floam_laser_mapping_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/floam_velodyne.rviz" />
    </group>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
